index.js

```diff
@@ -85,81 +91,113 @@ class GithubScm extends Scm {
-        const branchName = getInfo(result).branch;
-
-        // Do not convert branch name to lowercase
-        result = result.split('#')[0].toLowerCase().concat(`#${branchName}`);
-
-        return result;
-    }
 
     /**
-    * Get a users permissions on a repository
-    * @method _getPermissions
-    * @param  {Object}   config            Configuration
-    * @param  {String}   config.scmUrl     The scmUrl to get permissions on
-    * @param  {String}   config.token      The token used to authenticate to the SCM
-    * @return {Promise}
-    */
-    _getPermissions(config) {
-        const scmInfo = getInfo(config.scmUrl);
+     * Look up a repo by SCM URI
+     * @method lookupScmUri
+     * @param  {Object}     config Config object
+     * @param  {Object}     config.scmUri The SCM URI to look up relevant info
+     * @param  {Object}     config.token  Service token to authenticate with Github
+     * @return {Promise}                  Resolves to an object containing
+     *                                    repository-related information
+     */
+    lookupScmUri(config) {
+        const scmId = 1;
+        const scmUriParts = config.scmUri.split(':');
 
         return new Promise((resolve, reject) => {
             this.breaker.runCommand({
-                action: 'get',
+                action: 'getById',
                 token: config.token,
-                params: {
-                    user: scmInfo.user,
-                    repo: scmInfo.repo
-                }
+                params: { id: scmUriParts[scmId] }
             }, (error, data) => {
                 if (error) {
                     return reject(error);
                 }
 
-                return resolve(data.permissions);
+                const repoNameParts = data.full_name.split('/');
+
+                return resolve({
+                    branch: scmUriParts[PART_SCM_URI_BRANCH],
+                    host: scmUriParts[PART_SCM_URI_HOST],
+                    repo: repoNameParts[PART_REPO_NAME],
+                    user: repoNameParts[PART_OWNER_NAME]
+                });
             });
         });
     }
 
     /**
+    * Get a users permissions on a repository
+    * @method _getPermissions
+    * @param  {Object}   config            Configuration
+    * @param  {String}   config.scmUri     The scmUri to get permissions on
+    * @param  {String}   config.token      The token used to authenticate to the SCM
+    * @return {Promise}
+    */
+    _getPermissions(config) {
+        return this.lookupScmUri({
+            scmUri: config.scmUri,
+            token: config.token
+        }).then(scmInfo =>
+            new Promise((resolve, reject) => {
+                this.breaker.runCommand({
+                    action: 'get',
+                    token: config.token,
+                    params: {
+                        repo: scmInfo.repo,
+                        user: scmInfo.user
+                    }
+                }, (error, data) => {
```

Comment:
It shouldn't be too much change (since I changed it for the parseUrl PR) but I'm fine with another PR